### PR TITLE
Add trailing slash workaround to generic odata client

### PR DESCRIPTION
**Pull request by artemkovalyov**
## What Has Changed?

Add a workaround for a missing trailing slash on the resource path in case the service might need it.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
~~- [ ] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.~~

### CODE REVIEW

**Review by FrankEssenberger — COMMENTED** [review]
LGTM however I have the question if we want to promote the untyped client. It is a valid workaround but I would prefer to have a aligned solution for JS and Java e.g. some retry with the slash, an option to remove the stripping, a setter for the path URL which is taken as is... there are multiple ways to solve this. We should try to fix things and use the untyped client only as a last resort.

**Comment by FrankEssenberger** [comment]
Perhaps the Java colleagues can also comment on this.

**Comment by artemkovalyov** [comment]
It was my first thought, @FrankEssenberger as well. I agree we have to provide an "official workaround" read feature so that with or without trailing slash is possible on the type-safe client. I'll create a BLI for the Java team.

Having a workaround with the advanced client is a must before we release the nice way to do it.

**Comment by artemkovalyov** [comment]
I'll wait for the review from @Johannes-Schneider before merging this.

File: docs/java/features/odata/generic-odata-client.mdx [hunk 1]
```diff
@@ -234,3 +234,24 @@ ODataRequestFunction request =
...
 This is essentially what happens under the hood when you are using the generated type-safe client code.
 When we release a support for type-safe calls you can simple replace this snippets with the fluent API.
+
+### Adding Trailing Slash to the Service Path
+
+By default the SAP Cloud SDK strips all trailing slashes from a resource path.
+It's a standard behavior and works for 99% of productively available services.
+Sometimes services might complain or even fail if services path doesn't end with a slash `/`.
+
+- `https://host/servise-path/resource-path/?$filter...` - works
```
Comment (Johannes-Schneider) [inline docs/java/features/odata/generic-odata-client.mdx:244]:
```suggestion
- `https://host/service-path/resource-path/?$filter...` - works
```

File: docs/java/features/odata/generic-odata-client.mdx [hunk 2]
```diff
@@ -234,3 +234,24 @@ ODataRequestFunction request =
...
+- `https://host/servise-path/resource-path?$filter..` - doesn't work (default URI composition strategy by the SAP Cloud SDK)
```
Comment (Johannes-Schneider) [inline docs/java/features/odata/generic-odata-client.mdx:245]:
```suggestion
- `https://host/service-path/resource-path?$filter..` - doesn't work (default URI composition strategy in the SAP Cloud SDK)
```

File: docs/java/features/odata/generic-odata-client.mdx [hunk 3]
```diff
@@ -234,3 +234,24 @@ ODataRequestFunction request =
 
 This is essentially what happens under the hood when you are using the generated type-safe client code.
 When we release a support for type-safe calls you can simple replace this snippets with the fluent API.
+
+### Adding Trailing Slash to the Service Path
+
+By default the SAP Cloud SDK strips all trailing slashes from a resource path.
+It's a standard behavior and works for 99% of productively available services.
+Sometimes services might complain or even fail if services path doesn't end with a slash `/`.
```
Comment (Johannes-Schneider) [inline docs/java/features/odata/generic-odata-client.mdx:242]:
```suggestion
Sometimes, however, services might complain or even fail if a path doesn't end with a slash `/`.
```

File: docs/java/features/odata/generic-odata-client.mdx [hunk 4]
```diff
@@ -234,3 +234,24 @@ ODataRequestFunction request =
...
+
+- `https://host/servise-path/resource-path/?$filter...` - works
+- `https://host/servise-path/resource-path?$filter..` - doesn't work (default URI composition strategy by the SAP Cloud SDK)
+
+To workaround this situation you can use generic OData client
```
Comment (Johannes-Schneider) [inline docs/java/features/odata/generic-odata-client.mdx:247]:
```suggestion
To workaround this situation you can use generic OData client:
```

File: docs/java/features/odata/generic-odata-client.mdx [hunk 5]
```diff
@@ -234,3 +234,24 @@ ODataRequestFunction request =
...
+### Adding Trailing Slash to the Service Path
+
+By default the SAP Cloud SDK strips all trailing slashes from a resource path.
+It's a standard behavior and works for 99% of productively available services.
+Sometimes, however, services might complain or even fail if a path doesn't end with a slash `/`.
+
+- `https://host/service-path/resource-path/?$filter...` - works
+- `https://host/service-path/resource-path?$filter..` - doesn't work (default URI composition strategy in the SAP Cloud SDK)
+
+To workaround this situation you can use generic OData client:
```
Comment (Johannes-Schneider) [inline docs/java/features/odata/generic-odata-client.mdx:247]:
```suggestion
To workaround this situation you can use the generic OData client:
```

**Review by Johannes-Schneider — APPROVED** [review]
LGTM.

**Pull request merged**